src/main/java/com/aws/greengrass/mqttclient/MqttClient.java

```diff
@@ -979,6 +939,66 @@ protected int getNextClientIdNumber() {
         return 0;
     }
 
+    /**
+     * Create AwsIotMqttConnectionBuilder to generate an mqtt connection client.
+     *
+     * @param deviceConfiguration used to get the values needed for mqtt connection builder
+     * @param clientBootstrap client bootstrap values for mqtt connection builder
+     */
+    @SuppressWarnings("PMD.PreserveStackTrace")
+    public AwsIotMqttConnectionBuilder createMqttConnectionBuilder(DeviceConfiguration deviceConfiguration,
```

Comment:
pass in the security service as a parameter instead of a field of this class. Ideally this method should be static.